packages/store/src/IStore.sol

```diff
@@ -49,7 +49,10 @@ interface IStore {
 interface IStoreHook {
   function onSetRecord(uint256 table, bytes32[] memory key, bytes memory data) external;
 
-  function onSetField(uint256 table, bytes32[] memory key, uint8 schemaIndex, bytes memory data) external;
+  // Split onSetField into pre and post to simplify the implementation of hooks
```

Comment:
Without this hooks have to reimplement the StoreCore logic to replace a certain piece of bytes in the previous value with the new value. By splitting it up hooks can handle the previous value in `preSetField` and the new value in `postSetField` (see `ReverseMappingHook` as an example). This approach is less gas efficient than letting the hook compute the new value in memory, so we might change this during a gas optimization pass. Will add to #444.